src/builder.rs

```diff
@@ -625,5 +623,7 @@ const fn texture_format_size(texture_format: wgpu::TextureFormat) -> f32 {
 
         // 12x12 blocks, 16 bytes per block
         Astc { block: B12x12, channel: _ } => 9.0, // 12.0 * 12.0 / 16.0
+
+        NV12 => todo!(),
```

Comment:
This needs to be implemented. The size appears to be 3.

Reply:
How do you figure that? I'm finding conflicting figures when trying to get the answer. Some say it's 12 bits per pixel - the wgpu documentation says that it consists of two 8-bit planes.

Reply:
The WGPU docs say that it is two 8-bit planes, that's true, but the second plane is a _dual 8-bit channel_. It is 12-bits per pixel, there are just two pixels per "block". 24-bits total.

Reply:
Ok, how should I comment the size amount then?

Reply:
Side note: I could be totally wrong. But that's how I read it.

Reply:
So, I found this, which seems to be _the_ authoritative answer: https://github.com/gfx-rs/wgpu/blob/84ba4e5461fd1f305f8591cb086c40a488fbdb71/wgpu-hal/src/vulkan/conv.rs#L77

Which, if we lookup in Vulkan documentation at https://docs.vulkan.org/spec/latest/chapters/formats.html#formats-compatibility says:

> Block size 3 byte

And you can search for `VK_FORMAT_G8_B8R8_2PLANE_420_UNORM` to find the following prose description:

> `VK_FORMAT_G8_B8R8_2PLANE_420_UNORM` specifies an unsigned normalized multi-planar format that has an 8-bit G component in plane 0, and a two-component, 16-bit BR plane 1 consisting of an 8-bit B component in byte 0 and an 8-bit R component in byte 1. The horizontal and vertical dimensions of the BR plane are halved relative to the image dimensions, and each R and B value is shared with the G components for which \(\left\lfloor i_G \times 0.5 \right\rfloor = i_B = i_R\) and \(\left\lfloor j_G \times 0.5 \right\rfloor = j_B = j_R\). The location of each plane when this image is in linear layout can be determined via [vkGetImageSubresourceLayout](https://docs.vulkan.org/spec/latest/chapters/resources.html#vkGetImageSubresourceLayout), using `VK_IMAGE_ASPECT_PLANE_0_BIT` for the G plane, and `VK_IMAGE_ASPECT_PLANE_1_BIT` for the BR plane. This format only supports images with a width and height that is a multiple of two.

Reply:
Alright, I'll put it down as a 24-bit format then.

Reply:
```suggestion
        // 8-bit two-plane 4:2:0 YUV
        NV12 => 3.0, // (8.0 + 16.0) / 8.0
```

Reply:
```suggestion
        // 8-bit two-plane 4:2:0 YUV
        // The first plane consists of 8-bit G components. the second plane consists of 16-bit BR components. The resolution of the second plane is halved both vertically and horizontally.
        NV12 => 1.5, // (8.0 + 16.0 / 2.0 / 2.0) / 8.0
```

Reply:
Hi, sorry, could you update this again? I worked it out in my head, and your original finding of 12-bits per pixel is correct. It isn't 24-bits.

Reply:
Done.